src/transformers/models/bartpho/tokenization_bartpho.py

```diff
@@ -157,12 +157,22 @@ def __init__(
         self.sp_model.Load(str(vocab_file))
 
         # Load the reduced vocab
-        self.fairseq_tokens_to_ids = {"<s>": 0, "<pad>": 1, "</s>": 2, "<unk>": 3}
+
+        # Keep order of special tokens for backward compatibility
+        self.fairseq_tokens_to_ids = {
+            token: token_id
+            for token_id, token in enumerate(
+                dict.fromkeys(
+                    [str(bos_token), str(pad_token), str(eos_token), str(unk_token), str(sep_token), str(cls_token)]
+                ).keys()
+            )
+        }
```

Comment:
These are ancillary changes to the PR to remove values that were hardcoded in the tokenizer. They are nevertheless relevant for this PR in order to be able to re-generate the `monolingual_vocab_file` file (without risking to make mistakes) when saving.